api/v1alpha4/cluster_types.go

```diff
@@ -60,6 +60,64 @@ type ClusterSpec struct {
 	// for provisioning infrastructure for a cluster in said provider.
 	// +optional
 	InfrastructureRef *corev1.ObjectReference `json:"infrastructureRef,omitempty"`
+
+	// This encapsulates the topology for the cluster.
+	// +optional
+	Topology *Topology `json:"topology,omitempty"`
+}
+
+// Topology encapsulates the information of the managed resources.
+type Topology struct {
+	// The name of the ClusterClass object to create the topology.
+	Class string `json:"class,omitempty"`
+
+	// The kubernetes version of the cluster.
+	Version string `json:"version"`
+
+	// RolloutAfter performs a rollout of the entire cluster one component at a time [...]
+	// +optional
+	RolloutAfter *metav1.Time `json:"rolloutAfter,omitempty"`
+
+	// The information for the Control plane of the cluster.
+	ControlPlane ControlPlane `json:"controlPlane"`
+
+	// Worker encapsulates the different constructs that form the worker nodes
+	// for the cluster.
+	// +optional
+	Worker *Worker `json:"worker,omitempty"`
```

Comment:
```suggestion
	// The information for the Control plane of the cluster.
	ControlPlane ControlPlaneTopology `json:"controlPlane"`

	// Worker encapsulates the different constructs that form the worker nodes
	// for the cluster.
	// +optional
	Worker *WorkerTopology `json:"worker,omitempty"`
```

For the sake of consistency, we should probably suffix the tree of objects under `Topology` with that word. @CecileRobertMichon thoughts about it? We can also update the proposal in this PR

Reply:
I'm okay with it either way, I think it's fine like this too. Shouldn't it be `Workers` plural though?

If we make any changes to the proposed types, +1 to updating the proposal